htrc/torchlite/app.py

```diff
@@ -11,19 +14,29 @@
 from htrc.torchlite.filters import torchlite_stopword_filter, torchlite_stemmer, torchlite_lemmatizer
 from htrc.torchlite.middleware import TorchliteVersionHeaderMiddleware
 
-with open("config.yaml", mode="r", encoding="utf-8") as f:
-    config = safe_load(f)
+config_file: Optional[str] = os.getenv("TORCHLITE_CONFIG")
 
+if config_file is None:
+    raise OSError("Torchlite Configuration not found")
 
-def set_defaults(app: Torchlite, ef_api: Api) -> None:
-    if "featured_worksets_url" in config:
-        url = config["featured_worksets_url"]
-        if url:
-            r = requests.get(url)
-            if r.status_code == 200:
-                sample_workset_data = load(r.text, Loader=Loader)
-                for data in sample_workset_data:
-                    app.add_workset(**data)
+if config_file.startswith("http"):
+    resp: requests.Response = requests.get(config_file)
+    if resp.status_code == 200:
+        config = safe_load(resp.text)
+    else:
+        raise OSError(f"{config_file} not found")
```

Comment:
I don't think "not found" is accurate in all cases --- there could be other errors returned that don't mean the file doesn't exist (like let's say the SSL certificate expired). A better message would be "Could not load {config_file} - reason: {resp.status_code}"

An even better approach is described at the bottom of [this message](https://stackoverflow.com/a/16511493/1112620)

Reply:
Since the code is working, I'm going to merge this pr but make a ticket: https://github.com/htrc/torchlite-backend/issues/39